Modules/vc-module-cart/VirtoCommerce.CartModule.Web/Controllers/Api/CartModuleController.cs

```diff
@@ -0,0 +1,283 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+using VirtoCommerce.CartModule.Core;
+using VirtoCommerce.CartModule.Core.Model;
+using VirtoCommerce.CartModule.Core.Model.Search;
+using VirtoCommerce.CartModule.Core.Services;
+using VirtoCommerce.CoreModule.Core.Payment;
+using VirtoCommerce.CoreModule.Core.Shipping;
+using VirtoCommerce.Platform.Core.Caching;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.CartModule.Web.Controllers.Api
+{
+    [Route("api/carts")]
+    [EnableCors]
+    public class CartModuleController : Controller
+    {
+        private readonly IShoppingCartService _shoppingCartService;
+        private readonly IShoppingCartSearchService _searchService;
+        private readonly IShoppingCartBuilder _cartBuilder;
+
+        public CartModuleController(IShoppingCartService shoppingCartService, IShoppingCartSearchService searchService, IShoppingCartBuilder cartBuilder)
+        {
+            _shoppingCartService = shoppingCartService;
+            _searchService = searchService;
+            _cartBuilder = cartBuilder;
+        }
+
+        [HttpGet]
+        [Route("{storeId}/{customerId}/{cartName}/{currency}/{cultureName}/current")]
+        [ProducesResponseType(typeof(ShoppingCart), 200)]
```

Comment:
Need replace to strongly typed response ActionResult<T> everywhere in this controller